dds/DCPS/security/CryptoBuiltInImpl.cpp

```diff
@@ -1638,9 +1754,20 @@ bool CryptoBuiltInImpl::decode_serialized_payload(
     return false;
   }
 
+  if (security_debug.encdec) {
+    ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) {encdec} CryptoBuiltInImpl::decode_serialized_payload ")
+      ACE_TEXT("Sending DWCH is %u, Receiving DRCH is %u\n"),
+      sending_datawriter_crypto, receiving_datareader_crypto));
+  }
+
   ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
-  if (!writer_options_[sending_datawriter_crypto].payload_) {
+  if (!encrypt_options_[sending_datawriter_crypto].payload_) {
     plain_buffer = encoded_buffer;
+    if (security_debug.encdec) {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) {encdec} CryptoBuiltInImpl::decode_serialized_payload ")
+        ACE_TEXT("Sending datawriter should not be encrypting, returning input as plaintext\n"),
```

Comment:
The phrase "should not be encrypting" could be read as "it did encrypt even though it shouldn't have".  Could you reword this?